enginetest/queries/index_queries.go

```diff
@@ -4110,6 +4110,51 @@ var IndexQueries = []ScriptTest{
 			},
 		},
 	},
+	{
+		// MySQL allows creating multiple indexes over the same set of columns. This isn't generally
+		// useful, but some customers need this support. For example, generated migration code from
+		// Django can create cases that require this: https://github.com/dolthub/dolt/issues/8254
+		Name: "multiple indexes over same set of columns",
+		SetUpScript: []string{
+			"CREATE TABLE `t0` (`id` char(32) NOT NULL PRIMARY KEY, `col1` varchar(255) NOT NULL, `col2` varchar(255) NOT NULL);",
```

Comment:
Interestingly, create table statements with multiple identical unique keys work, but are overwritten once we have an alter

```sql
tmp/main> create table t (i int, unique index (i), unique index(i), unique index(i));
tmp/main*> show create table t;
+-------+------------------------------------------------------------------+
| Table | Create Table                                                     |
+-------+------------------------------------------------------------------+
| t     | CREATE TABLE `t` (                                               |
|       |   `i` int,                                                       |
|       |   UNIQUE KEY `i` (`i`),                                          |
|       |   UNIQUE KEY `i_2` (`i`),                                        |
|       |   UNIQUE KEY `i_3` (`i`)                                         |
|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin |
+-------+------------------------------------------------------------------+
1 row in set (0.00 sec)

tmp/main*> alter table t add unique index (i);
tmp/main*> show create table t;
+-------+------------------------------------------------------------------+
| Table | Create Table                                                     |
+-------+------------------------------------------------------------------+
| t     | CREATE TABLE `t` (                                               |
|       |   `i` int,                                                       |
|       |   UNIQUE KEY `i_4` (`i`)                                         |
|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin |
+-------+------------------------------------------------------------------+
1 row in set (0.00 sec)

```

Reply:
oof! more wacky MySQL behavior. Good find. I may skip this behavior unless a customer explicitly asks for it.

Reply:
Ah... I misread this comment the first time through; sorry about that! I see now you meant this was a bug in Dolt's behavior. 

I was able to repro this behavior with the current version of `dolt` and I confirmed that it is fixed in the latest version. I opened https://github.com/dolthub/go-mysql-server/pull/2637 to add an additional test case to cover this one.